general_considerations.md

```diff
@@ -390,6 +559,12 @@ hasher could construct a block with invalid transactions, but this would be
 easily discovered if that hasher ever mined a block, and his shares could
 invalidated.
 
+[Comment: I proposed that this should be done before receiving miners include a
+sending minder's shares in the DAG. The receiving miners include a back pointer
+to a received share, thus including it in the DAG. Once enough miners have a
+back pointer to a share, it is kind of buried in the DAG and thus gets
+rewarded. That is the general intuition behind reaching agreement with DAGs.]
+
```

Comment:
I think you're saying we should do ex-post facto verification of a share's block? I agree but the devil is in the details. The required bandwidth is considerable and I've been thinking about spot-checking and sharing TX set differences using Wuille's algorithm.

e.g. dynamically organize shareholders into random quorums that verify each other's share-blocks. If the quorum size is 4, each miner needs to ensure 4 specific other shareholders receive enough data to verify his share. 3 of those 4 must post a signature (possibly mined into a bead) witnessing that they validated the share.

I assume that all transactions in a share-block are available to verifiers. If there are non-mempool transactions they must be shared with the share-block-validators. (and then I'd say the share validators should broadcast those transactions!)